integration-tests/client/test/core.test.ts

```diff
@@ -0,0 +1,355 @@
+
+        // Get initial balances - only track what we can reliably query
+        const { balance: { amount: strideInitialStrideBalance } = { amount: "0" } } =
+            await stridejs.query.cosmos.bank.v1beta1.balance({
+                address: stridejs.address,
+                denom: USTRD,
+            });
+
+        const { balance: { amount: strideInitialAtomBalance } = { amount: "0" } } =
+            await stridejs.query.cosmos.bank.v1beta1.balance({
+                address: stridejs.address,
+                denom: ATOM_DENOM_ON_STRIDE,
+            });
+
+        const gaiaInitialAtomBalance = await gaiajs.query.bank.balance(gaiajs.address, UATOM);
+
+        console.log("Initial balances:");
+        console.log(`Stride USTRD: ${strideInitialStrideBalance}`);
+        console.log(`Stride ATOM: ${strideInitialAtomBalance}`);
+        console.log(`Gaia ATOM: ${gaiaInitialAtomBalance.amount}`);
+
+        // Perform IBC transfers
+        console.log("Transferring USTRD from Stride to Gaia...");
+        await ibcTransfer({
+            client: stridejs,
+            sourceChain: "STRIDE",
+            destinationChain: "GAIA",
+            coin: `${transferAmount}${USTRD}`,
+            sender: stridejs.address,
+            receiver: gaiajs.address,
+        });
+
+        console.log("Transferring ATOM from Gaia to Stride...");
+        await ibcTransfer({
+            client: gaiajs,
+            sourceChain: "GAIA",
+            destinationChain: "STRIDE",
+            coin: `${transferAmount}${UATOM}`,
+            sender: gaiajs.address,
+            receiver: stridejs.address,
+        });
+
+        // Wait a bit for transfers to complete
+        await sleep(5000);
+
+        // Get final balances
+        const { balance: { amount: strideFinalStrideBalance } = { amount: "0" } } =
+            await stridejs.query.cosmos.bank.v1beta1.balance({
+                address: stridejs.address,
+                denom: USTRD,
+            });
+
+        const { balance: { amount: strideFinalAtomBalance } = { amount: "0" } } =
+            await stridejs.query.cosmos.bank.v1beta1.balance({
+                address: stridejs.address,
+                denom: ATOM_DENOM_ON_STRIDE,
+            });
+
+        const gaiaFinalAtomBalance = await gaiajs.query.bank.balance(gaiajs.address, UATOM);
+
+        console.log("Final balances:");
+        console.log(`Stride USTRD: ${strideFinalStrideBalance}`);
+        console.log(`Stride ATOM: ${strideFinalAtomBalance}`);
+        console.log(`Gaia ATOM: ${gaiaFinalAtomBalance.amount}`);
+
+        // Calculate and verify balance changes
+        const strideBalanceDiff = BigInt(strideInitialStrideBalance) - BigInt(strideFinalStrideBalance);
+        const strideAtomBalanceDiff = BigInt(strideFinalAtomBalance) - BigInt(strideInitialAtomBalance);
+        const gaiaAtomBalanceDiff = BigInt(gaiaInitialAtomBalance.amount) - BigInt(gaiaFinalAtomBalance.amount);
```

Comment:
nit: make them all (final - initial) since it's easier to track the sign that way, and make the denom explicit in the variable name
```suggestion
        const strideStrdBalanceDiff = BigInt(strideFinalStrdBalance) - BigInt(strideInitialStrdBalance);
        const strideAtomBalanceDiff = BigInt(strideFinalAtomBalance) - BigInt(strideInitialAtomBalance);
        const gaiaStrdBalanceDiff = BigInt(gaiaFinalStrdBalance) - BigInt(gaiaInitialStrdBalance);
        const gaiaAtomBalanceDiff = BigInt(gaiaFinalAtomBalance) - BigInt(gaiaInitialAtomBalance);
```

Reply:
done. thanks.